### PR TITLE
Fix filtering of node ids for TransportNodesAction

### LINKED ISSUE #18618: Filtered list of nodes IDs in TransportNodesAction isn't actually considered
In the 2.x branch, the `TransportNodesAction` class defines two methods:

```
    protected String[] filterNodeIds(DiscoveryNodes nodes, String[] nodesIds) {
        return nodesIds;
    }

    protected String[] resolveNodes(NodesRequest request, ClusterState clusterState) {
        return clusterState.nodes().resolveNodesIds(request.nodesIds());
    }
```

which basically return all the nodes in the cluster (`resolveNodes`) and just the nodes IDs received as parameter (`filterNodeIds`), respectively.

The constructor, though, does the following:

```
            String[] nodesIds = resolveNodes(request, clusterState);
            this.nodesIds = filterNodeIds(clusterState.nodes(), nodesIds);
            ImmutableOpenMap<String, DiscoveryNode> nodes = clusterState.nodes().nodes();
            this.nodes = new DiscoveryNode[nodesIds.length];
            for (int i = 0; i < nodesIds.length; i++) {
                this.nodes[i] = nodes.get(nodesIds[i]);
            }
            this.responses = new AtomicReferenceArray<>(this.nodesIds.length);
        }
```

So, the list of nodes - `this.nodes[i] = nodes.get(nodesIds[i]);` - is being built considering the result of `resolveNodes()` method (meaning all the nodes), not on the filtered list of nodes (the result of `filterNodeIds` method), which should be the desired behavior.

### PR DESCRIPTION
**Pull request by brwe**
Don't mix up member variables  with local variables
in constructor.

closes #18618

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java [hunk 1]
```diff
@@ -176,12 +176,12 @@ protected NodeResponse nodeOperation(NodeRequest request, Task task) {
             this.request = request;
             this.listener = listener;
             ClusterState clusterState = clusterService.state();
-            String[] nodesIds = resolveNodes(request, clusterState);
-            this.nodesIds = filterNodeIds(clusterState.nodes(), nodesIds);
-            ImmutableOpenMap<String, DiscoveryNode> nodes = clusterState.nodes().getNodes();
+            String[] nodesIdCandidates = resolveNodes(request, clusterState);
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java]: maybe we should just get rid of this local variable and write the next line:
```
nodesIds = filterNodeIds(clusterState.nodes(), resolveNodes(request, clusterState));
```

File: core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java [hunk 2]
```diff
@@ -176,12 +176,12 @@ protected NodeResponse nodeOperation(NodeRequest request, Task task) {
...
+            nodesIds = filterNodeIds(clusterState.nodes(), nodesIdCandidates);
+            ImmutableOpenMap<String, DiscoveryNode> nodesCandidates = clusterState.nodes().getNodes();
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java]: We should also get rid of this local variable (The map is not needed as `DiscoveryNodes` has a method `get()`).

We could write `this.nodes[i] = clusterState.nodes().get(nodesIds[i]);` a few lines below.

If `clusterState.nodes()` is too verbose, we can extract that one into a local variable.

**Comment by ywelsch** [comment]
Made some suggestions to improve readability of the code. If you don't agree, feel free to push. LGTM.

**Comment by brwe** [comment]
@ywelsch I removed the local variables.

File: core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java [hunk 3]
```diff
@@ -176,12 +176,10 @@ protected NodeResponse nodeOperation(NodeRequest request, Task task) {
...
             this.listener = listener;
             ClusterState clusterState = clusterService.state();
-            String[] nodesIds = resolveNodes(request, clusterState);
-            this.nodesIds = filterNodeIds(clusterState.nodes(), nodesIds);
-            ImmutableOpenMap<String, DiscoveryNode> nodes = clusterState.nodes().getNodes();
+            nodesIds = filterNodeIds(clusterState.nodes(), resolveNodes(request, clusterState));
             this.nodes = new DiscoveryNode[nodesIds.length];
             for (int i = 0; i < nodesIds.length; i++) {
-                this.nodes[i] = nodes.get(nodesIds[i]);
+                this.nodes[i] = clusterState.nodes().getNodes().get(nodesIds[i]);
```
Comment (ywelsch) [inline core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java]: please remove the additional `getNodes()` here :-)

Comment (brwe) [inline core/src/main/java/org/elasticsearch/action/support/nodes/TransportNodesAction.java]: ok, done

**Comment by ywelsch** [comment]
LGTM

**Comment by brwe** [comment]
Thanks for the review @ywelsch !

**Pull request merged**